rewriter/rewriter.cc

```diff
@@ -154,7 +155,15 @@ class Rewriterer {
     }
 
     unique_ptr<ast::Expression> postTransformSend(core::MutableContext ctx, unique_ptr<ast::Send> send) {
-        return InterfaceWrapper::run(ctx, std::move(send));
+        if (auto expr = InterfaceWrapper::run(ctx, send.get())) {
+            return expr;
+        }
+
+        if (auto expr = SelfNew::run(ctx, send.get())) {
+            return expr;
+        }
+
+        return send;
```

Comment:
Huh. Why do we have two different places to register DSL passed on Send’s? E.g. the prop DSL and all the other DSLs on sends except InterfaceWrapper are above. Does SelfNew need to be here? Could we move all of them to be in one place?

Reply:
I'll move them both; I agree that it's strange having two ways to do the same thing.

Reply:
Following up, we have both because the typecase in ClassDef only handles sends that are in the RHS of the ClassDef. I'll add a comment that describes why we have the postTransformSend case as well.